Source/AFError.swift

```diff
@@ -125,17 +121,50 @@ public enum AFError: Error {
         case inputFileReadFailed(at: URL)
         case stringSerializationFailed(encoding: String.Encoding)
         case jsonSerializationFailed(error: Error)
-        case propertyListSerializationFailed(error: Error)
         case invalidEmptyResponse(type: String)
     }
 
+    public enum ServerTrustFailureReason {
+        public struct Output {
+            public let host: String
+            public let trust: SecTrust
+            public let status: OSStatus
+            public let result: SecTrustResultType
+
+            init(_ host: String, _ trust: SecTrust, _ status: OSStatus, _ result: SecTrustResultType) {
+                self.host = host
+                self.trust = trust
+                self.status = status
+                self.result = result
+            }
+        }
+        case unknown(host: String)
+        case noRequiredEvaluator(host: String)
+        case noCertificatesFound
+        case noPublicKeysFound
+        case policyApplicationFailed(trust: SecTrust, policy: SecPolicy, status: OSStatus)
+        case settingAnchorCertificatesFailed(status: OSStatus, certificates: [SecCertificate])
+        case revocationPolicyCreationFailed
+        case defaultEvaluationFailed(output: Output)
+        case hostValidationFailed(output: Output)
+        case revocationCheckFailed(output: Output, options: RevocationTrustEvaluator.Options)
+        case certificatePinningFailed(host: String, trust: SecTrust, pinnedCertificates: [SecCertificate], serverCertificates: [SecCertificate])
+        case publicKeyPinningFailed(host: String, trust: SecTrust, pinnedKeys: [SecKey], serverKeys: [SecKey])
+    }
+
     case explicitlyCancelled
     case invalidURL(url: URLConvertible)
     case parameterEncodingFailed(reason: ParameterEncodingFailureReason)
     case multipartEncodingFailed(reason: MultipartEncodingFailureReason)
     case responseValidationFailed(reason: ResponseValidationFailureReason)
     case responseSerializationFailed(reason: ResponseSerializationFailureReason)
-    case certificatePinningFailed
+    case serverTrustEvaluationFailed(reason: ServerTrustFailureReason)
+}
+
+public extension Error {
+    public var asAFError: AFError? {
+        return self as? AFError
+    }
 }
```

Comment:
I've added a doc comment, since it's public.

Reply:
I completely agree that users will be more likely to try to cast Error to `AFError` to implement custom error handling logic. I'm just not sold yet that it warrants a separate API to do that. How exactly does this make it easier? Are you thinking that it's easier because there's now an auto-complete option to do it?

I'm not super opposed to this addition or anything, I'm just not sure why you see it as being "easier" to use `asAFError` instead of `as? AFError`. I'm also not sure why using the property is better than having casts everywhere. Is that because we'd be able to refactor it if needed in the future? Just trying to understand...

Reply:
A property is more discoverable, easier to use (autocomplete, like you said), and saves the consideration of `as?` for users who aren't familiar with it. It's the typically suggested pattern for dealing with specific error types you have to cast from `Error`. Like I said, I envision using it a lot just in our own testing, even if users don't.